p2p/src/net/types/services.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use serialization::{Decode, Encode};
+
+#[derive(Eq, PartialEq, Clone, Copy, Debug, Encode, Decode, Default)]
+pub struct Service(u32);
```

Comment:
Why not an enum? Something like

```
pub enum PublicFlags {
    Transactions = 0x01,
    Blocks = 0x02,
    PeerAddresses = 0x04,
}
```

Then you can define how services are concatenated.

There's no particular problem with what you did, but I just find this nicer. Maybe I'm missing something.

Reply:
The only issue I see with the current implementation is that it's possible to have two "arms" with the same value.

Reply:
Thanks for the hint, that is much better.